### PR TITLE
Added analyzer and code fixer for IHttpContextAccessor

### PR DESCRIPTION
**Pull request by shahramshahram**

### CODE REVIEW

**Review by lunchin — APPROVED** [review]

File: src/EpiSourceUpdater/EpiHttpContextBaseAccessorCodeFixProvider.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+        private static async Task<Document> ReplaceParameterTypeAsync(Document document, ClassDeclarationSyntax classDeclarationSyntax, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var updatedClassDeclaration = UpdateMemberField(classDeclarationSyntax);
+            var updatedClassDeclarationByCostr = UpdateConstructor(updatedClassDeclaration);
+            var newRoot = root!.ReplaceNode(classDeclarationSyntax, updatedClassDeclarationByCostr);
+
+            // Return document with transformed tree.
+            var updatedDocument = document.WithSyntaxRoot(newRoot);
```
Comment (jbearfoot) [inline src/EpiSourceUpdater/EpiHttpContextBaseAccessorCodeFixProvider.cs:70]: There is now an extension method document.AddUsingIfMissingAsync that you can use to add using statements

File: src/EpiSourceUpdater/EpiHttpContextBaseAccessorCodeFixProvider.cs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+        private static ClassDeclarationSyntax UpdateMemberField(ClassDeclarationSyntax classDecSyntax)
+        {
+            var updatedMembers = new List<MemberDeclarationSyntax>();
+            foreach (var m in classDecSyntax.Members)
+            {
+                if (m is FieldDeclarationSyntax field && field.Declaration.Type.ToString().Equals(EpiHttpContextBaseAccessorAnalyzer.HttpContextBaseParameterType, StringComparison.OrdinalIgnoreCase))
+                {
+                    var svcAccessorHttpContextBaseMember = m as FieldDeclarationSyntax;
+                    //var variableDeclaration = svcAccessorHttpContextBaseMember.Declaration.WithType(SyntaxFactory.ParseTypeName("IHttpContextAccessor").WithTrailingTrivia(SyntaxFactory.Whitespace(" "))).WithVariables(svcAccessorHttpContextBaseMember.Declaration.Variables).WithLeadingTrivia(SyntaxFactory.Whitespace(" "));
```
Comment (jbearfoot) [inline src/EpiSourceUpdater/EpiHttpContextBaseAccessorCodeFixProvider.cs:93]: I guess this out-commented code can be removed?

File: src/EpiSourceUpdater/Resources.resx [hunk 3]
```diff
@@ -153,6 +153,15 @@
   <data name="EpiDisplayChannelTitle" xml:space="preserve">
     <value>CMS display channels will be updated</value>
   </data>
+  <data name="EpiHttpContextBaseDescription" xml:space="preserve">
+    <value>ServiceAccessor&lt;HttpContextBase&gt; in construcor will changed to HttpContextAccessor.</value>
```
Comment (jbearfoot) [inline src/EpiSourceUpdater/Resources.resx:157]: Spelling error construcor => constructor

File: src/EpiSourceUpdater/EpiHttpContextBaseAccessorCodeFixProvider.cs [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+                    }
+                }
+
+                var updateConstructor =  (constructorSyntax as ConstructorDeclarationSyntax).WithParameterList(SyntaxFactory.ParameterList(SyntaxFactory.SeparatedList(parameters)));
+                updatedClassDeclaration = updatedClassDeclaration.ReplaceNode(constructorSyntax, updateConstructor);
+            }
+            return updatedClassDeclaration;
+        }
+
+        private static CompilationUnitSyntax AddUsing(CompilationUnitSyntax compilationRoot, string namespaceToAdd)
```
Comment (jbearfoot) [inline src/EpiSourceUpdater/EpiHttpContextBaseAccessorCodeFixProvider.cs:120]: I guess this method can be removed now?

**Review by jbearfoot — APPROVED** [review]

**Pull request merged**